Makefile

```
@@ -1,28 +1,58 @@
 BUILDDIR := ./_build
+FETCH_DATA_COMPLETE := $(BUILDDIR)/data/.complete
+CREATE_HTML_COMPLETE := $(BUILDDIR)/plugins/.complete
 
-.PHONY: all prep fetch-data create-html
+.PHONY: all clean prep fetch-data create-html serve-local
 
 all: prep fetch-data create-html
 
-prep:
-	@echo "Preparing build directory..."
+define check-venv
+	@echo "Checking if virtual environment is activated..." && \
+	if [ -z "$$VIRTUAL_ENV" ]; then \
```

I think this should generally work for conda as well, otherwise it only checks venv.

```suggestion
	if [ -z "$$VIRTUAL_ENV" ] && [ -z "$$CONDA_DEFAULT_ENV"]; then \
```